packages/big-design/src/components/Button/spec.tsx

```diff
@@ -188,24 +182,51 @@ test('render only icon only with left and right icons button', () => {
   expect(screen.getAllByTestId('icon-only')).toHaveLength(1);
 });
 
-test('hides content when isLoading is true', () => {
-  const plusIcon = <AddIcon role="icon" aria-hidden="false" />;
-  const { rerender } = render(
-    <Button isLoading={false} iconLeft={plusIcon}>
-      Button
-    </Button>,
-  );
-  const visibleContent = screen.getByRole('icon');
+describe('isLoading', () => {
+  test('render loading button', () => {
+    render(<Button isLoading={true}>Button</Button>);
 
-  expect(visibleContent).toBeInTheDocument();
+    expect(screen.getByRole('button')).toMatchSnapshot();
+  });
 
-  rerender(
-    <Button isLoading={true}>
-      <strong>Button</strong>
-    </Button>,
-  );
+  test('prevents on click', () => {
+    const mockClickHandler = jest.fn();
+    render(
+      <Button isLoading={true} onClick={mockClickHandler}>
+        Button
+      </Button>,
+    );
+
+    userEvent.click(screen.getByRole('button'), undefined, { skipPointerEventsCheck: true });
```

Comment:
This is a bit superfluous, checking that the button is disabled should be enough, but 🤷